blog/2024-04-08-reductstore-vs-mongodb/index.md

```diff
@@ -0,0 +1,254 @@
+
+For 10 KB chunks, MongoDB writes at a rate of 258 blobs per second, while ReductStore records a slightly lower rate at 223 blobs per second—a 14% decrease. However, when reading the same chunk size, ReductStore shows a marginal speed advantage with an increase of 4%.
+
+As chunk sizes escalate to 100 KB and beyond, up to 10 MB, ReductStore demonstrates significantly better performance over MongoDB. Write speeds improve from +36% for the former to +75% for the latter size tier. Similarly, read operations see improvements ranging from +34% up to +100%, highlighting ReductStore's efficiency in handling larger binary payloads.
+
+| Chunk Size | Operation | MongoDB, blob/s | ReductStore, blob/s | ReductStore, %    |
+|------------|-----------|-----------------|---------------------|-------------------|
+| 10 KB      | Write     | 258             | 223                 | -14%              |
+|            | Read      | 187             | 195                 | +4%               |
+| 100 KB     | Write     | 145             | 197                 | +36%              |
+|            | Read      | 108             | 145                 | +34%              |
+| 1 MB       | Write     | 32              | 52                  | +63%              |
+|            | Read      | 20              | 29                  | +45%              |
+| 10 MB      | Write     | 4               | 7                   | +75%              |
+|            | Read      | 2               | 4                   | +100%             |
+
+The observed performance trends underscore that while both systems are capable of managing binary data effectively, their throughput varies depending on blob sizes—with ReductStore gaining an edge as size increases.
+
+## Additional Criteria
+
+When selecting a database for blob storage, other criteria than performance must be taken into account:
+
+- **Retention policy**: both ReductStore and MongoDB offer distinct retention strategies. ReductStore uses disk utilization-based policies, essential for edge computing and IoT. MongoDB's time-series collections, on the other hand, can automatically purge documents after a set period of time, useful for managing data lifecycle when dealing with non-blob data such as numerical values or logs in time series scenarios.
+
+- **Replication features**: Replication in ReductStore is append-only, with label-based filtering options. This feature can contribute to an effective data reduction strategy by selectively replicating relevant blobs. MongoDB, on the other hand, offers replication sets for high availability and redundancy, ensuring data integrity and fault tolerance in distributed environments.
+
+- **Cost considerations**: ReductStore's pricing model is based on storage which aligns well with retention policies based on data volume for a predictable cost structure. MongoDB's pricing is based on instance size and usage, which can be more complex to estimate.
```

Comment:
Just checked sortly, their license is very similar to AGPL. You can use it for free as long as you don't provide MongoDB as service. If you do it, you should open your source code.

Reply:
@atimin I added a point "License Consideration" to be checked